android/src/main/java/com/swmansion/reanimated/NativeMethodsHelper.java

```diff
@@ -57,9 +57,24 @@ public static void scrollTo(View view, double argX, double argY, boolean animate
 
     if (animated) {
       if (horizontal) {
-        ((ReactHorizontalScrollView) view).smoothScrollTo((int) x, (int) y);
+        ReactHorizontalScrollView finalView = view;
+        view.post(
+            new Runnable() {
+              @Override
+              public void run() {
+                finalView.smoothScrollTo(x, y);
+              }
+            });
```

Comment:
It looks like code duplication with lines 71-77. Could we move it outside the `if` condition?

Reply:
I think we can't, we have to call view.post in the same context we define the view variable.